### ISSUE TITLE
Add basic summary docs for all types/modules

### ISSUE THREAD

**Issue by dsyme** [fsprojects/FSharpx.Async#3]
Great work to extract this as an independent thing

Some types/modules are missing a "summary" entry, see http://fsprojects.github.io/FSharpx.Async/reference/index.html

**Comment by forki** [comment]
was about time... https://twitter.com/sforkmann/status/569867563670372352

**Comment by tpetricek** [comment]
I was thinking of doing something about the Async libraries too :-).

Though I would quite like to rename the library and not use the "FsharpX" label. I don't have a concrete name proposal (yet!), but I think some catchy name would be nicer than "FSharpX.Something" or "FSharp.Something".

There is also a [bunch of samples I wrote here](https://github.com/tpetricek/FSharp.AsyncExtensions/tree/master/samples), which could be turned into tutorials.

**Comment by forki** [comment]
Just to clarify. I extracted this because people asked me to do it. I don't
really know how to go forward and I'm would be happy if someone else could
take this.

**Comment by tpetricek** [comment]
Also, it would be great to get some feedback on this from @rachelreese and @eulerfx, because I think they have been using the libraries at work. So, they might even have updated versions!

**Comment by tpetricek** [comment]
@forki Sure thing - I'm very happy this is happening! Would be nice if Rachel and Lev were able to join us on this as maintainers :-). I should be able to contribute some docs.

**Comment by eulerfx** [comment]
I'd be more than happy to contribute docs. Should I document the source code itself or create a page in the wiki?

**Comment by tpetricek** [comment]
@eulerfx Thanks for the PRs & ideas! Are these based on some code that you already implemented & used? (If so, I'd be in favor of accepting most of it.. but will add some comments. If not, then I don't have to hurry with the API review that much :-)).

**Comment by forki** [comment]
looks a bit like:
![tumblr_inline_njhdyhcrqv1raprkq](https://cloud.githubusercontent.com/assets/57396/6351847/246acce0-bc3f-11e4-8cee-e49eef1b5c15.gif)

**Comment by eulerfx** [comment]
@tpetricek Yes, this is all implemented, but I'm concerned about consistent naming. What would be a good way to resolve this in tandem? I can post gists of the code and discuss from there? 

As an example, consider the following operation on `Async`:

``` fsharp
val strength (a:'a) (b:Async<'b>) : Async<'a * 'b>
```

The name `strength` is somewhat accurate in the [mathematical](http://ncatlab.org/nlab/show/tensorial+strength) sense, but quite cryptic otherwise. This operation is useful however for propagating inputs to arrows as shown [here](https://github.com/fsprojects/FSharpx.Async/issues/10). Similarly, `AsyncSeq.groupBy` as described [here](https://github.com/fsprojects/FSharpx.Async/issues/8) could use some naming attention.

I don't think it would be a good idea to just dump a bunch of code in and then consider renaming later as that will break backwards compatibility. Maybe an experimental project would be helpful?

EDIT

I'm making it more complicated than it has to be. We can just use PRs for this discussion.
